### PR TITLE
Finished CRUD Functions for Research Groups

### LINKED ISSUE #21: As an Admin I want to be able to add a research group so if there will be a new one it can be added easily.
**Acceptance criteria:**
GIVEN On the Admin Panel,
WHEN Click on the button add research group,
THEN the admin will be able to add a research group by fill in the inputs.
WHEN not all field is filled,
THEN the admin will see what he is missing.

### PR DESCRIPTION
**Pull request by avanheeren**
Added Edit and Delete CRUD for Research Groups,
fixed merge conflicts

### CODE REVIEW

File: app/Http/Controllers/ResearchGroupController.php [hunk 1]
```diff
@@ -22,9 +22,9 @@ public function index()
      *
      *
      */
-    public function create()
+    public function create(ResearchGroup $researchGroup)
```
Comment (Koreyoshi159) [inline app/Http/Controllers/ResearchGroupController.php]: ??? why researchGroup ?

File: app/Http/Controllers/ResearchGroupController.php [hunk 2]
```diff
@@ -33,11 +33,10 @@ public function create()
      * @param  \Illuminate\Http\Request  $request
      * @return \Illuminate\Http\Response
      */
-    public function store(Request $request)
+    public function store(ResearchGroup $researchGroup, Request $request)
```
Comment (Koreyoshi159) [inline app/Http/Controllers/ResearchGroupController.php]: Same for this one

File: resources/views/admin.blade.php [hunk 3]
```diff
@@ -101,12 +97,17 @@
...
-                                                    <button class="has-text-right"><a href="\">Edit</a>
+                                                    <button class="has-text-right">
+                                                        <a href="{{ route('research_groups.edit', $researchGroup) }}">Edit</a>
                                                     </button>
                                                 </td>
                                                 <td>
-                                                    <button class="has-text-right"><a
-                                                            href="\">Delete</a></button>
+                                                    <form method="POST"
+                                                          action="{{route('research_groups.destroy', $researchGroup)}}">
```
Comment (Koreyoshi159) [inline resources/views/admin.blade.php:106]: maybe on one line but only if you want to xD

File: resources/views/researchGroups/create.blade.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+            <div class="field is-horizontal">
+                <div class="field-label">
+                    <!-- Left empty for spacing -->
+                </div>
+                <div class="field-body">
+                    <div class="field">
+                        <div class="control">
+                            <button class="button is-info" type="submit">
+                                <p>Create!</p>
```
Comment (awqfrxx) [inline resources/views/researchGroups/create.blade.php]: p element in button??

File: resources/views/researchGroups/edit.blade.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+            <div class="field is-horizontal">
+                <div class="field-label">
+                    <!-- Left empty for spacing -->
+                </div>
+                <div class="field-body">
+                    <div class="field">
+                        <div class="control">
+                            <button class="button is-info" type="submit">
+                                <p>Edit</p>
```
Comment (awqfrxx) [inline resources/views/researchGroups/edit.blade.php]: again..

File: app/Models/ResearchGroup.php [hunk 6]
```diff
@@ -8,6 +8,7 @@
 class ResearchGroup extends Model
 {
     use HasFactory;
+    protected $guarded = [];
 
     protected $fillable = ['name'];
```
Comment (Koreyoshi159) [inline app/Models/ResearchGroup.php]: There is already a fillable variable why you added guarded

File: resources/views/researchGroups/create.blade.php [hunk 7]
```diff
@@ -1,12 +1,12 @@
...
 
 @section('content')
-    <section class="hero  is-medium  is-bold">
+    <section class="hero is-medium is-bold">
         <div class="header-img">
             <img src="/image/sdg-header-en.png"/>
         </div>
     </section>
-    <form method="POST" action="{{route('research_groups.store')}}">
+    <form method="POST" action="{{route('research_groups.store') }}">
```
Comment (awqfrxx) [inline resources/views/researchGroups/create.blade.php]: white space 😠

Comment (Koreyoshi159) [reply resources/views/researchGroups/create.blade.php]: A space after {{

**Review by awqfrxx — CHANGES_REQUESTED** [review]
good job!

**Review by Koreyoshi159 — APPROVED** [review]
Well done

**Review by awqfrxx — APPROVED** [review]

**Pull request merged**